src/lirc.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2022 Sean Young <sean@mess.org>
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+#include "defs.h"
+#include <linux/lirc.h>
+#include "xlat/lirc_features.h"
+#include "xlat/lirc_modes.h"
+
+int
+lirc_ioctl(struct tcb *const tcp, const unsigned int code,
+	   const kernel_ulong_t arg)
+{
+	int value;
+
+	if (_IOC_DIR(code) == _IOC_READ && entering(tcp))
+		return 0;
+
+	if (umove_or_printaddr(tcp, arg, &value))
+		return RVAL_IOCTL_DECODED;
+
+	tprint_arg_next();
+
+	switch (code) {
+	case LIRC_GET_FEATURES:
+		printflags(lirc_features, value, "LIRC_CAN_???");
+		break;
+	case LIRC_GET_SEND_MODE:
+	case LIRC_GET_REC_MODE:
+	case LIRC_SET_SEND_MODE:
+	case LIRC_SET_REC_MODE:
+		printxval(lirc_modes, value, "LIRC_MODE_???");
+		break;
+	default:
+		PRINT_VAL_D(value);
+		break;
+	}
```

Comment:
Since `value` is obtained from tracee's memory, its printing should be wrapped with
```
tprint_indirect_begin();
...
tprint_indirect_end();
```

See e.g. `decode_fs_ioc_flags` as an example.

Also, `tprint_arg_next();` should precede any printing, this includes `umove_or_printaddr`.

Reply:
good catch, done